relay-metrics/src/aggregation.rs

```diff
@@ -1874,6 +1878,10 @@ impl AggregatorService {
         let capped_batches =
             CappedBucketIter::new(buckets.into_iter(), self.config.max_flush_bytes);
 
+        if capped_batches.unused_data.is_some() {
```

Comment:
This uses an internal field of the utility iterator. I would prefer if we implement one of two alternatives:
1. Move the error log into the iterator in place of setting `unused_data`.
2. Implement a method `pub fn is_something(&self) -> bool` that performs this check.

Since this is really an implementation/config bug, I think we don't have to consider this edge case in the public signature of the iterator, so I'd slightly prefer option 1.

What are your thoughts, @olksdr?

Reply:
That makes sense and I like option 1 much better.

Reply:
@jan-auer  I've removed the extra field `unused_data` and just log the error in: 

https://github.com/getsentry/relay/blob/fb98c0917acf106443abe2da537a0615eeb16d23/relay-metrics/src/aggregation.rs#L1325-L1330